### PR TITLE
[Model] Add LongCat-Image support

### PR DESCRIPTION
**Pull request by e1ijah1**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose

Add support for LongCat-Image #220 

## Test Plan

```shell
python3 examples/offline_inference/text_to_image/text_to_image.py --model meituan-longcat/LongCat-Image/ --prompt '一个年轻的亚裔女性，身穿黄色针织衫，搭配白色项链。她的双手放在膝盖上，表情恬静。背景是一堵粗糙的砖墙，午后的阳光温暖地洒在她身上，营造出一种宁静而温馨的氛围。镜头采用中距离视角，突出她的神态和服饰的细节。光线柔和地打在她的脸上，强调她的五官和饰品的质感，增加画面的层次感与亲和力。整个画面构图简洁，砖墙的纹理与阳光的光影效果相得益彰，突显出人物的优雅与从容。' --width 1024 --height 1024 --num_inference_steps 50 --num_images_per_prompt 1 --output longcat_image_output.png
```

## Test Result

longcat_image_output

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by e1ijah1** [comment]
@codex review

**Comment by e1ijah1** [comment]
cc @ZJY0516 @hsliuustc0106 PTAL

File: vllm_omni/diffusion/models/longcat_image/system_prompt.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (ZJY0516) [inline vllm_omni/diffusion/models/longcat_image/system_prompt.py:1]: Please add links to the original sources of these prompts

File: vllm_omni/diffusion/models/longcat_image/longcat_image_transformer.py [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+            head_size=self.head_dim,
+            total_num_heads=self.heads,
+            disable_tp=True,
+            bias=bias,
+        )
+
+        if not self.pre_only:
+            self.to_out = nn.ModuleList([])
+            self.to_out.append(torch.nn.Linear(self.inner_dim, self.out_dim, bias=out_bias))
+            self.to_out.append(nn.Dropout(dropout))
```
Comment (ZJY0516) [inline vllm_omni/diffusion/models/longcat_image/longcat_image_transformer.py]: Dropout is not applied during the inference phase by default. Please remove this

**Review by ZJY0516 — COMMENTED** [review]
please fix pre-commit and doc error

File: vllm_omni/diffusion/models/longcat_image/longcat_image_transformer.py [hunk 3]
```diff
@@ -0,0 +1,513 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+
+from collections.abc import Iterable
+from typing import Any, Optional, Union
+
+import torch
+import torch.nn as nn
+from diffusers.models.attention import FeedForward
```
Comment (ZJY0516) [inline vllm_omni/diffusion/models/longcat_image/longcat_image_transformer.py]: Could you please implement FeedForward in this file? We may want to extend it to support tp later.

**Review by ZJY0516 — COMMENTED** [review]
please fix the doc build error

File: vllm_omni/diffusion/models/longcat_image/longcat_image_transformer.py (hunk 3 above)
Comment (e1ijah1) [reply vllm_omni/diffusion/models/longcat_image/longcat_image_transformer.py]: FeedForward has been added, PTAL

File: tests/e2e/offline_inference/test_t2i_model.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 os.environ["VLLM_TEST_CLEAN_GPU_MEMORY"] = "1"
 
 
-models = ["Tongyi-MAI/Z-Image-Turbo", "riverclouds/qwen_image_random"]
+models = ["Tongyi-MAI/Z-Image-Turbo", "riverclouds/qwen_image_random", "meituan-longcat/LongCat-Image"]
```
Comment (ZJY0516) [inline tests/e2e/offline_inference/test_t2i_model.py]: Could we remove this test temporarily? We have time limit per test now.

File: vllm_omni/diffusion/models/longcat_image/system_prompt.py
Comment (SamitHuang) [inline vllm_omni/diffusion/models/longcat_image/system_prompt.py:1]: import it from `diffusers/pipelines/longcat_image/system_messages.py`?

**Review by hsliuustc0106 — APPROVED** [review]
lgtm
